enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="15"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="20"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="25"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="30"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="35"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="40"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="45"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="50"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="55"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="60"} 1
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="+Inf"} 1
+cluster_installation_create_sum{platform="aws",result="success",version="4.2",os="linux"} 30
+cluster_installation_create_count{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+The example above represents a single sample. This sample indicates that a Linux user successfully installed a 4.2 AWS cluster, which took between 30 and 35 minutes to complete installation. 
+
+###### Label Values
+* `result` - provides limited context to the outcome of an attempted install:
+    - `ProvisioningFailed`
+    - `BootstrapFailed`
+    - `APIFailed`
+    - `OperatorsDegraded`
+    - `Success`
+* `platform` - currently we deploy to 8 platforms: `aws`, `azure`, `gcp,` `libvirt`, `none`, `openstack`, `ovirt`, and `vsphere`. The list will grow. 
+* `os` - You can find all the available combinations of [Go OS and arch values](https://github.com/golang/go/blob/master/src/go/build/syslist.go) with `go tool dist list`. At this stage I assume we only want to colllect operating systems and some of the values are unlikely/impossible:
+  - `linux`
+  - `darwin` - this could be rewritten to `macos` on the installer client if we prefer
+  - `bsd` - this would encompass the `freebsd`, `netbsd`, and `openbsd` values
+  - `windows` - future proofing? 
+  - `other` - everything else not captured above
+* `version` - represents minor releases; assuming we do not backport metrics, `4.5` will be the only value if we make it into the next release and we could expect roughly 20 values over the next five years.
+* `le` - the standard Prometheus label for histogram buckets, this represents how long a command took complete. The current proposal is to have a 5 minute step starting at 15 minutes and maxing at 60.
+
+##### cluster_installation_waitfor
+UPI installs utilize the wait-for command, which would be tracked with a similar histogram:
```

Comment:
When an install fails, Hive uses `wait-for` commands to (1) check whether bootstrapping succeeded and, in some cases, (2) wait longer for cluster initialization to complete. Maybe we could add a flag for the `wait-for` commands to not do metrics gathering so that Hive's use cases do not skew results.